docs/source/llms/tracing/index.rst

```diff
@@ -0,0 +1,615 @@
+
+As a comparison between the fluent and the client APIs for tracing, the figure below illustrates the differences in complexity between the two APIs, 
+with the fluent API being more concise and the recommended approach if your tracing use case can support using the higher-level APIs.
+
+.. figure:: ../../_static/images/llms/tracing/fluent-vs-client-tracing.png
+    :alt: Fluent vs Client APIs
+    :width: 60%
+    :align: center
+
+This section will cover how to initiate traces using these fluent APIs.
+
+Initiating a Trace
+^^^^^^^^^^^^^^^^^^
+
+In this section, we will explore different methods to initiate a trace using MLflow's fluent APIs. These methods allow you to add tracing 
+functionality to your code with minimal modifications, enabling you to capture detailed information about the execution of your functions and workflows.
+
+Trace Decorator
+###############
+
+The trace decorator allows you to automatically capture the inputs and outputs of a function by simply adding the :py:func:`@mlflow.trace <mlflow.trace>` decorator 
+to its definition. This approach is ideal for quickly adding tracing to individual functions without significant changes to your existing code.
+
+.. code-block:: python
+
+    import mlflow
+
+    # Create a new experiment to log the trace to
+    mlflow.set_experiment("Tracing Demo")
+
+
+    # Mark any function with the trace decorator to automatically capture input(s) and output(s)
+    @mlflow.trace
+    def some_function(x, y, z=2):
+        return x + (y - z)
+
+
+    # Invoking the function will generate a trace that is logged to the active experiment
+    some_function(2, 4)
+
+You can add additional metadata to the tracing decorator as follows:
+
+.. code-block:: python
+
+    @mlflow.trace(name="My Span", span_type="func", attributes={"a": 1, "b": 2})
+    def my_func(x, y):
+        return x + y
+
+When adding additional metadata to the trace decorator constructor, these additional components will be logged along with the span entry within 
+the trace that is stored within the active MLflow experiment.
+
+**What is captured?**
+
+If we navigate to the MLflow UI, we can see that the trace decorator automatically captured the following information, in addition to the basic
+metadata associated with any span (start time, end time, status, etc):
+
+- **Inputs**: In the case of our decorated function, this includes the state of all input arguments (including the default `z` value that is applied).
+- **Response**: The output of the function is also captured, in this case the result of the addition and subtraction operations.
+- **Trace Name**: The name of the decorated function.
+
+.. figure:: ../../_static/images/llms/tracing/trace-demo-1.png
+    :alt: Trace UI - simple use case
+    :width: 100%
+    :align: center
+
+**What about Error handling?**
+
+If an `Exception` is raised during processing of a trace-instrumented operation, an indication will be shown within the UI that the invocation was not 
```

Comment:
nit: Could we add a brief sentence that says exception details are recorded within `span.events`? (Or perhaps we should wait until UI for checking events is ready?)

Reply:
I think adding the sentence is a good idea.